rfcs/datepicker-component.md

```diff
@@ -0,0 +1,148 @@
+# DatePicker Component
+
+Represents control to work and pick date, time value or a range of values
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => <DatePicker />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {DatePicker} from 'baseui/datepicker';
+
+export default () => {
+  return <DatePicker
+    initialState={{
+      value: [new Date(2018, 1, 15), new Date(2018, 8, 25)]
+    }}
+    isRange={true}
+    dateFormat={'MM/DD/YY'}
+    minDate={new Date(2000, 1, 1)}
+    maxDate={new Date(2020, 1, 1)}
+    monthRange={2}
+    onChange={(event, value)=>{console.log(value)}}
+    overrides={{
+      MonthGridDay: props => <div style={{
+        color: props.$inRange ? 'red' : 'blue',
+        borderRadius: '5px',
+      }}>{props.$value}</div>,
+    }}
+    />
+}
+```
+
+## Exports
+
+* `DatePicker`
+* `StatefulDatePicker`
+* `StatefulDatePickerContainer`
+* `StyledRoot`
+* `StyledDateLabel`
+* `StyledDateInput`
+* `StyledMonthAndYearBar`
+* `StyledMonthSelect`
+* `StyledYearSelect`
+* `StyledMonthGrid`
+* `StyledMonthGridLabel`
+* `StyledMonthGridDay`
+* `StyledQuickSelect`
+
+## `DatePicker` API
+
+* `isRange: boolean` - Optional
+  If set, datepicker is to select 2 dates range, so there are two date input and `value` comes as Array of two dates, min and max. Otherwise only one date is selected. Default is `false`
+* `dateFormat: string` - Optional
+  Date format that is accepted by `Date.parse` function. RFC2822 tools.ietf.org/html/rfc2822#section-3.3 standard. Default is `DD/MM/YYYY`
+* `value: Date | Array<Date>` - Optional
+  Selected `Date`. If `isRange` is set, `value` is an array of 2 values.
+* `minDate: Date` - Optional
+  minimum date available for selection. Default is `Jan 01 1970`
+* `maxDate: Date` - Optional
+  maximum date available for selection. Default is current date + 3 years.
+* `disabled: boolean`:
+  Disable control from being changed
+* `monthRange: number`
+  Amount of minth grids to be shown. Accepts number of visible months(mostly 1 or 2). Default is `1`
+* `overrides: {Root: (props: {[string]: any}) => React$Node, DateLabel: (props: {[string]: any}) => React$Node,
+ DateInput: (props: {[string]: any}) => React$Node, MonthAndYearBar: (props: {[string]: any}) => React$Node,
+  MonthSelect: (props: {[string]: any}) => React$Node, YearSelect: (props: {[string]: any}) => React$Nod,
+  MonthGrid: (props: {[string]: any}) => React$Node, MonthGridLabel: (props: {[string]: any}) => React$Node,
+  MonthGridDay: (props: {[string]: any}) => React$Node, QuickSelect: (props: {[string]: any}) => React$Node}`
```

Comment:
Please fix the formatting on this, make it an object with line breaks and spacing.

Reply:
ok